biohansel/cli.py

```diff
@@ -0,0 +1,281 @@
+@click.option('-t', '--threads',
+              type=int,
+              default=1,
+              help='Number of parallel threads to run analysis (default=1)')
+@click.argument('files', type=click.Path(exists=True), nargs=-1)
+def subtype(scheme,
+            scheme_name,
+            scheme_metadata,
+            paired_reads,
+            input_fasta_genome_name,
+            input_directory,
+            output_summary_path,
+            output_tile_results,
+            output_simple_summary_path,
+            force,
+            json_output,
+            min_kmer_freq,
+            max_kmer_freq,
+            low_coverage_threshold,
+            max_missing_tiles,
+            min_ambiguous_tiles,
+            low_coverage_warning,
+            max_intermediate_tiles,
+            threads,
+            files):
+    """Subtype microbial genomes using SNV targeting k-mer subtyping schemes.
+
+    Includes subtyping schemes for:
+
+    - Salmonella enterica spp. enterica serovar Heidelberg
+
+    - Salmonella enterica spp. enterica serovar Enteritidis
+
+    Developed by Geneviève Labbé, James Robertson, Peter Kruczkiewicz, Marisa Rankin, Matthew Gopez, Chad R. Laing, Philip Mabon, Kim Ziebell, Aleisha R. Reimer, Lorelee Tschetter, Gary Van Domselaar, Sadjia Bekal, Kimberley A. MacDonald, Linda Hoang, Linda Chui, Danielle Daignault, Durda Slavic, Frank Pollari, E. Jane Parmley, Philip Mabon, Elissa Giang, Lok Kan Lee, Jonathan Moffat, Marisa Rankin, Joanne MacKinnon, Roger Johnson, John H.E. Nash.
+    """
+    does_file_exist(output_simple_summary_path, force)
+    does_file_exist(output_summary_path, force)
+    does_file_exist(output_tile_results, force)
+    scheme_fasta = get_scheme_fasta(scheme)
+    scheme_subtype_counts = subtype_counts(scheme_fasta)
+
+    subtyping_params = init_subtyping_params(**locals())
+    input_contigs, input_reads = collect_inputs(**locals())
+    if len(input_contigs) == 0 and len(input_reads) == 0:
+        no_files_exception = click.UsageError('No input files specified!')
+        click.secho('Please see -h/--help for more info', err=True)
+        raise no_files_exception
+    df_md = None
+    if scheme_metadata:
+        df_md = read_metadata_table(scheme_metadata)
+
+    subtype_results = []  # type: List[Tuple[Subtype, pd.DataFrame]]
+    if len(input_contigs) > 0:
+        contigs_results = subtype_contigs_samples(input_genomes=input_contigs,
+                                                  scheme=scheme,
+                                                  scheme_name=scheme_name,
+                                                  subtyping_params=subtyping_params,
+                                                  scheme_subtype_counts=scheme_subtype_counts,
+                                                  n_threads=threads)
+        logging.info('Generated %s subtyping results from %s contigs samples', len(contigs_results), len(input_contigs))
```

Comment:
Perhaps f strings would be good to keep things consistent, just a small thing

Reply:
Yes and the f-string syntax is easier to read than the alternatives. Changed in c2fb9cf815c8a8923fecda61671ffc851782db0c